security/vulnerability-management.md

```diff
@@ -0,0 +1,85 @@
+- **Waive**: Rule to ignore a vulnerability for a specific amount of time
+- **Severity Levels**: Classification of vulnerabilities based on their potential impact.
+
+## 4. Roles and Responsibilities
+- **Developers:** Responsible for writing secure code, conducting initial testing, and fixing identified vulnerabilities.
+- **Project Managers:** Responsible for ensuring that vulnerabilities are addressed and review waivers for vulnerabilities in accordance with
+this policy
+
+## 5. Classification of Vulnerabilities
+Vulnerabilities will be classified into four severity levels:
+- **Critical**: Immediate action required. These vulnerabilities can lead to complete system compromise, significant data breach/loss.
+- **High**: Must be addressed as soon as possible. These can lead to major security breaches.
+- **Medium**: Should be fixed within a reasonable timeframe. These can potentially be exploited but are not as severe.
+- **Low**: Minor issues that should be resolved in the regular course of development.
+
+## 6. Identification of Vulnerabilities
+The following mechanism will be used identify potential vulnerabilities
+- **IDE Tools**: Developer's IDE will be the first gate to identify potential vulnerabilities through static analysis or dependency scan
+- **Manual Code Reviews**: Developers will conduct regular manual code reviews to identify and raise any potential security concerns.
+- **Automated Scanning**: Automated tools will run on schedule or on commit actions, performing scans and triggering notifications
+- **External Audits**: Engage third-party security experts to conduct periodic security assessments.
+
+## 7. Remediation Process
+1. **Identification**: Identify the vulnerability and whether it impacts a public-facing application.
+2. **Assessment**: Assess the impact of the vulnerability if it were exploited and determine the severity.
+3. **Prioritization**: Prioritization of the investigation and potential remediation depending on the severity. In some cases the developers
+might be able to determine that their implementation is not exploitable instead of removing the vulnerablity
+4. **Remediation Plan**: A remediation plan is developed, including a timeline for addressing the vulnerability. For critical vulnerabilities
+some actions might need to be taken (such as temporarily disable a service)
+5. **Fix Implementation**: Developers implement the fixes as per the remediation plan. For high and critical vulnerabilities, immediate action
+is taken and a hotfix release might be required.
+6. **Verification**: The security team verifies the fix to ensure the vulnerability is effectively mitigated.
+7. **Deployment**: The fix is deployed following standard deployment procedures.
+
+## 8. Timelines for Remediation
+*Note: The following timelines represent what VisitScotland aims for but might not be currently representative as not all the tools and
+processes might be in place*
```

Comment:
Could this be reworded, I'm not sure this is entirely true.

Reply:
Hello Brian thank you for all the tips, I really appreciate that.

Regarding to this comment, What I tried to convey is that, we are not going to apply this timing strictly but we will introduce this in an iterative way. 

So for example, I think once we have this fully setup we will aim to deal with critical vulnerabilities within the current cycle unless it is a "major critical vulnerability" However, for Open-Source projects we might want to aim to the timelines mentioned in this section